File: go/commands/prefix/public/ping.go
package public

import (
	"fmt"
	"net/http"
	"time"

	"github.com/RealMtrx/Discord-Handler-Go/bot"
	"github.com/bwmarrin/discordgo"
)

func RegisterPrefixPing(b *bot.Bot) {
	b.PrefixCommands["ping"] = &bot.PrefixCommand{
		Name: "ping",
		Handler: func(s *discordgo.Session, m *discordgo.MessageCreate, args []string) {
			sent, _ := s.ChannelMessageSend(m.ChannelID, "⏳ Loading...")

			restStart := time.Now()
			client := &http.Client{Timeout: 5 * time.Second}
			client.Get("https://discord.com/api/v10/users/@me")
			restEnd := time.Now()

			botLatency := sent.Timestamp.Sub(m.Timestamp).Milliseconds()
			wsLatency := s.HeartbeatLatency().Milliseconds()
			restLatency := restEnd.Sub(restStart).Milliseconds()

			embed := &discordgo.MessageEmbed{
				Title: "🏓 Pong!",
				Color: 0x00FF00,
				Fields: []*discordgo.MessageEmbedField{
					{Name: "🤖 Bot Latency", Value: fmt.Sprintf("`%dms`", botLatency), Inline: true},
					{Name: "📡 WebSocket Latency", Value: fmt.Sprintf("`%dms`", wsLatency), Inline: true},
					{Name: "🌐 REST API Latency", Value: fmt.Sprintf("`%dms`", restLatency), Inline: true},
				},
				Timestamp: time.Now().Format(time.RFC3339),
			}

			s.ChannelMessageEditEmbed(m.ChannelID, sent.ID, embed)
		},
	}
}

File: src/Commands/Prefix/Public/ping.ts
import { EmbedBuilder } from "discord.js";
import { REST } from "@discordjs/rest";
import config from "../../../config.js";
import type { ExtendedClient, PrefixCommand } from "../../../types.js";

const rest = new REST({ version: "10" }).setToken(config.token);

export default {
  dir: "Public",
  name: "ping",
  description: "Show bot latency and API response times",

  run: async (client: ExtendedClient, message: any, args: string[]) => {
    const sent = await message.reply("⏳ Loading...");

    const restStart = Date.now();
    await rest.get("/users/@me");
    const restEnd = Date.now();

    const embed = new EmbedBuilder()
      .setTitle("🏓 Pong!")
      .setColor("Green")
      .addFields(
        {
          name: "🤖 Bot Latency",
          value: `\`${sent.createdTimestamp - message.createdTimestamp}ms\``,
          inline: true,
        },
        { name: "📡 WebSocket Latency", value: `\`${client.ws.ping}ms\``, inline: true },
        { name: "🌐 REST API Latency", value: `\`${restEnd - restStart}ms\``, inline: true }
      )
      .setTimestamp();

    sent.edit({ content: "", embeds: [embed] });
  },
} satisfies PrefixCommand;
